Tests/PublisherTests/Mocks/MockResolutionPolicyFactory.swift

```diff
@@ -4,7 +4,9 @@ import AblyAssetTrackingCore
 class MockResolutionPolicyFactory: ResolutionPolicyFactory {
     var resolutionPolicy: MockResolutionPolicy?
     func createResolutionPolicy(hooks: ResolutionPolicyHooks, methods: ResolutionPolicyMethods) -> ResolutionPolicy {
-        resolutionPolicy =  MockResolutionPolicy(hooks: hooks, methods: methods)
+        if resolutionPolicy == nil {
```

Comment:
- So `createResolutionPolicy` doesn't always create a `resolutionPolicy`? That's confusing/ bad name
- Why do we re-use it, is it really expensive? Or even worse, are we sharing state between different calls of `createResolutionPolicy`? I think we should avoid this singleton pattern if possible

Reply:
182e59433c22a24e407b597f08e716a30bc22648

Reply:
We don't need a field though, do we? It can just be returned from the function immediately? In tests, just save a reference to `resolutionPolicy` returned by `createResolutionPolicy`. I think that would be a nice simplification.

Reply:
this is not my idea, and since I'm not 100% sure that this change will not affect the test, I don't want to change it. This should be done here https://github.com/ably/ably-asset-tracking-swift/issues/185#issuecomment-923656235